pkg/pipelines/upgrade_cluster.go

```diff
@@ -34,7 +34,7 @@ func NewUpgradeClusterPipeline(runtime *common.KubeRuntime) error {
 	m := []module.Module{
 		&precheck.NodePreCheckModule{},
 		&precheck.ClusterPreCheckModule{},
-		&confirm.UpgradeConfirmModule{},
+		&confirm.UpgradeConfirmModule{Skip: runtime.Arg.SkipConfirmCheck},
```

Comment:
This variable hasn't been set, you need to set this variable on https://github.com/kubesphere/kubekey/blob/b5a41a021d49cb1bf90d2e2c8a3f042b87e7863a/cmd/ctl/upgrade/upgrade.go#L84

Reply:
ok